src/main/scala-3/com/sksamuel/scapegoat/Plugin.scala

```diff
@@ -0,0 +1,113 @@
+package com.sksamuel.scapegoat
+
+import java.io.File
+
+import com.sksamuel.scapegoat.io.IOUtils
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Contexts.{ctx, Context}
+import dotty.tools.dotc.plugins.{PluginPhase, StandardPlugin}
+import dotty.tools.dotc.reporting.Diagnostic
+import dotty.tools.dotc.reporting.Diagnostic.{Error, Info, Warning}
+import dotty.tools.dotc.transform.PatternMatcher
+import dotty.tools.dotc.typer.TyperPhase
+import dotty.tools.dotc.util.NoSourcePosition
+
+class ScapegoatPlugin extends StandardPlugin {
+
+  override def name: String = "scapegoat"
+
+  override def description: String = "scapegoat compiler plugin"
+
+  override val optionsHelp: Option[String] = Some(Configuration.optionsHelp)
+
+  override def init(options: List[String]): List[PluginPhase] = {
+    val config = Configuration.fromPluginOptions(options)
+    new ScapegoatPhase(config, Inspections.inspections) :: Nil
+  }
+
+}
+
+class ScapegoatPhase(var configuration: Configuration, override val inspections: Seq[Inspection])
+    extends PluginPhase
+    with ScapegoatBasePlugin {
+
+  import tpd.*
+
+  private[scapegoat] var feedback: Option[FeedbackDotty] = None
+
+  override def phaseName: String = "scapegoat"
+
+  override val runsAfter: Set[String] = Set(TyperPhase.name)
+
+  override val runsBefore: Set[String] = Set(PatternMatcher.name)
+
+  override def run(using Context): Unit = {
```

Comment:
todo: Need to double check this flow. Seems to write reports far too often in the CI check, is this running per file?